### PR TITLE
Use apt-mirror to mirror Ubuntu repos

### PR DESCRIPTION
**Pull request by cbosdo**
To enable cloning the Ubuntu repositories, edit the apt-mirror.list
file.

### CODE REVIEW

File: salt/mirror/apt-mirror.list [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+set mirror_path    /srv/mirror/
+set defaultarch    amd64
+set nthreads       20
+
+##
+## Sources
+##
+
+# deb http://archive.ubuntu.com/ubuntu xenial main
```
Comment (moio) [inline salt/mirror/apt-mirror.list]: Idea for the future: make this configurable with a variable on the mirror module. The same should happen for other very heavy repos such as RES/CentOS.

File: salt/repos/mirror.sls [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% if grains.get('role') == 'mirror' %}
+
+# We need that repository for apt-mirror
+suse_manager_devel_repo:
```
Comment (moio) [inline salt/repos/mirror.sls]:
```suggestion
tools_repo:
```

File: salt/mirror/apt-mirror.list (hunk 1 above)
Comment (cbosdo) [reply salt/mirror/apt-mirror.list]: Sounds cool indeed

File: salt/repos/mirror.sls (hunk 2 above)
Comment (cbosdo) [reply salt/repos/mirror.sls]: Fixed

File: salt/mirror/apt-mirror.list (hunk 1 above)
Comment (cbosdo) [reply salt/mirror/apt-mirror.list]: Implemented

**Comment by cbosdo** [comment]
Also fixed a few dependency problems in the mirror state found while testing

**Comment by cbosdo** [comment]
Works like a charm with and without the ubuntu repos to mirror

**Comment by cbosdo** [comment]
yeah, I wasn't too comfortable adding an `s` to Ubuntu. Fixed now

**Pull request merged**